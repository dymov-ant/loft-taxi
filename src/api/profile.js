import { api } from "./index"
import { AUTH_TOKEN } from "../constants"

export const getProfile = () => {
  return api.get("/card", {
    params: {token: localStorage.getItem(AUTH_TOKEN)}
  })
}

export const setProfile = (cardData) => {
  return api.post("/card", {...cardData, token: localStorage.getItem(AUTH_TOKEN)})
}