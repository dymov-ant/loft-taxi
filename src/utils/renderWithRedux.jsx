import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import { createMemoryHistory } from 'history';
// eslint-disable-next-line import/no-extraneous-dependencies
import configureStore from 'redux-mock-store';

const history = createMemoryHistory();
const mockStore = configureStore();

export const renderWithRedux = (component, initialState = {}) => {
  const store = mockStore(initialState);
  return render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  );
};
