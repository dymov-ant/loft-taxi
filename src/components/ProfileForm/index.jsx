import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useFormik } from 'formik';
import DateFnsUtils from '@date-io/date-fns';
import { Button, makeStyles, TextField, Typography } from '@material-ui/core';
import { KeyboardDatePicker, MuiPickersUtilsProvider } from '@material-ui/pickers';
import InputMask from 'react-input-mask';
import { Link } from 'react-router-dom';
import { cardSchema } from '../../utils/validationsSchemas';
import { profileActions } from '../../store/actions/profile';
import { ROUTE_NAMES } from '../../router';
import styles from './profileForm.module.scss';
import Spinner from '../Spinner';
import SavedCard from '../SavedCard';

const useStyles = makeStyles((theme) => ({
  input: {
    margin: theme.spacing(1, 0),
    // eslint-disable-next-line
    '@media (max-width:767px)': {
      margin: theme.spacing(0, 0, 0.5),
    },
  },
}));

const ProfileForm = () => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const isLoading = useSelector((state) => state.profile.isLoading);
  const card = useSelector((state) => state.profile.card);
  const isSuccessSubmitting = useSelector((state) => state.profile.isSuccessSubmitting);
  const initialValues = card
    ? { ...card }
    : {
        cardName: '',
        cardNumber: '',
        expiryDate: new Date(),
        cvc: '',
      };

  const formik = useFormik({
    initialValues,
    validationSchema: cardSchema,
    onSubmit: (values) => {
      dispatch(profileActions.setProfile(values));
    },
  });

  useEffect(
    () => () => {
      dispatch(profileActions.successSubmitting(false));
    },
    [dispatch],
  );

  return (
    <div className={styles.form__wrapper}>
      <h1 className={styles.form__title}>Профиль</h1>
      <p className={styles.form__text}>
        {isSuccessSubmitting
          ? 'Платёжные данные обновлены. Теперь вы можете заказывать такси.'
          : 'Введите платежные данные'}
      </p>
      {isSuccessSubmitting ? (
        <>
          <Button variant="contained" color="primary" component={Link} to={ROUTE_NAMES.MAP}>
            Перейти на карту
          </Button>
        </>
      ) : (
        <>
          <form className={styles.form__formWrapper} noValidate>
            <div className={styles.form__form}>
              <div className={styles.form__row}>
                <TextField
                  label="Имя владельца"
                  type="text"
                  fullWidth
                  className={classes.input}
                  inputProps={{ style: { textTransform: 'uppercase' } }}
                  name="cardName"
                  value={formik.values.cardName}
                  onChange={formik.handleChange}
                  error={formik.touched.cardName && Boolean(formik.errors.cardName)}
                  helperText={formik.touched.cardName && formik.errors.cardName}
                />
              </div>
              <div className={styles.form__row}>
                <InputMask
                  mask="9999 9999 9999 9999"
                  value={formik.values.cardNumber}
                  onChange={formik.handleChange}
                  maskChar={null}
                >
                  {(props) => (
                    <TextField
                      {...props}
                      label="Номер карты"
                      type="text"
                      fullWidth
                      className={classes.input}
                      name="cardNumber"
                      error={formik.touched.cardNumber && Boolean(formik.errors.cardNumber)}
                      helperText={formik.touched.cardNumber && formik.errors.cardNumber}
                    />
                  )}
                </InputMask>
              </div>
              <div className={styles.form__row}>
                <div className={styles.form__col}>
                  <MuiPickersUtilsProvider utils={DateFnsUtils}>
                    <KeyboardDatePicker
                      label="MM/YY"
                      format="MM/yy"
                      minDate={new Date()}
                      variant="inline"
                      openTo="year"
                      views={['year', 'month']}
                      autoOk
                      className={classes.input}
                      name="expiryDate"
                      value={formik.values.expiryDate}
                      onChange={(val) => formik.setFieldValue('expiryDate', val)}
                      error={formik.touched.expiryDate && Boolean(formik.errors.expiryDate)}
                      helperText={formik.touched.expiryDate && formik.errors.expiryDate}
                    />
                  </MuiPickersUtilsProvider>
                </div>
                <div className={styles.form__col}>
                  <InputMask mask="999" value={formik.values.cvc} onChange={formik.handleChange} maskChar={null}>
                    {(props) => (
                      <TextField
                        {...props}
                        label="CVC"
                        type="text"
                        fullWidth
                        className={classes.input}
                        name="cvc"
                        value={formik.values.cvc}
                        onChange={formik.handleChange}
                        error={formik.touched.cvc && Boolean(formik.errors.cvc)}
                        helperText={formik.touched.cvc && formik.errors.cvc}
                      />
                    )}
                  </InputMask>
                </div>
              </div>
            </div>
            <div className={styles.form__cardWrapper}>
              {card ? (
                <SavedCard cardNumber={card.cardNumber} expiryDate={card.expiryDate} />
              ) : (
                <Typography variant="h6" align="center">
                  Нет сохраненных карт
                </Typography>
              )}
            </div>
          </form>
          {isLoading ? (
            <Spinner />
          ) : (
            <Button variant="contained" color="primary" type="submit" onClick={formik.handleSubmit}>
              Сохранить
            </Button>
          )}
        </>
      )}
    </div>
  );
};

export default ProfileForm;
