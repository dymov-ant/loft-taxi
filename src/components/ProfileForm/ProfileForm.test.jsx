import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react';
import ProfileForm from './index';
import { renderWithRedux } from '../../utils/renderWithRedux';
import { profileActions } from '../../store/actions/profile';

describe('ProfileForm component', () => {
  const initialState = {
    profile: {
      isLoading: false,
      isSuccessSubmitting: false,
      card: null,
    },
  };
  const cardData = {
    id: 'testId',
    cardNumber: '1231 2312 3123 2131',
    expiryDate: '2022-10-31T19:00:00.000Z',
    cardName: 'Test Name',
    cvc: '123',
  };

  it('ProfileForm renders', () => {
    const { container } = renderWithRedux(<ProfileForm />, initialState);

    expect(container).toBeInTheDocument();
  });

  it('ProfileForm submit', async () => {
    initialState.profile.card = cardData;
    const { getByTestId } = renderWithRedux(<ProfileForm />, initialState);
    const setProfileSpy = jest.spyOn(profileActions, 'setProfile');
    const button = getByTestId('button');
    fireEvent.click(button);

    await waitFor(() => expect(setProfileSpy).toBeCalled());
  });

  it('ProfileForm snapshot', () => {
    const profileForm = renderWithRedux(<ProfileForm />, initialState);

    expect(profileForm).toMatchSnapshot();
  });

  it('ProfileForm with card snapshot', () => {
    initialState.profile.card = cardData;
    const profileForm = renderWithRedux(<ProfileForm />, initialState);

    expect(profileForm).toMatchSnapshot();
  });
});
