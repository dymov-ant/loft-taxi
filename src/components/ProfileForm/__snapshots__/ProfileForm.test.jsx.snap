// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileForm component ProfileForm snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form__wrapper"
      >
        <h1
          class="form__title"
        >
          Профиль
        </h1>
        <p
          class="form__text"
        >
          Введите платежные данные
        </p>
        <form
          class="form__formWrapper"
          novalidate=""
        >
          <div
            class="form__form"
          >
            <div
              class="form__row"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-3 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Имя владельца
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    name="cardName"
                    style="text-transform: uppercase;"
                    type="text"
                    value="Test Name"
                  />
                </div>
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-3 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Номер карты
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    name="cardNumber"
                    type="text"
                    value="1231 2312 3123 2131"
                  />
                </div>
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="form__col"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-input-3"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    MM/YY
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      name="expiryDate"
                      type="text"
                      value="11/22"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form__col"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-input-3 MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    CVC
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="cvc"
                      type="text"
                      value="123"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form__cardWrapper"
          >
            <div
              class="card"
            >
              <div
                class="card__top"
              >
                <img
                  alt="лого"
                  class="card__logo"
                  src="card-logo.svg"
                />
                <p
                  class="card__date"
                >
                  11/22
                </p>
              </div>
              <p
                class="card__number"
              >
                <span>
                  1231
                </span>
                <span>
                  2312
                </span>
                <span>
                  3123
                </span>
                <span>
                  2131
                </span>
              </p>
              <div
                class="card__bottom"
              >
                <img
                  alt="чип"
                  class="card__schema"
                  src="card-schema.svg"
                />
                <img
                  alt="mastercard"
                  class="card__mastercard"
                  src="master-card.svg"
                />
              </div>
            </div>
          </div>
        </form>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="button"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Сохранить
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form__wrapper"
    >
      <h1
        class="form__title"
      >
        Профиль
      </h1>
      <p
        class="form__text"
      >
        Введите платежные данные
      </p>
      <form
        class="form__formWrapper"
        novalidate=""
      >
        <div
          class="form__form"
        >
          <div
            class="form__row"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-3 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                Имя владельца
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="cardName"
                  style="text-transform: uppercase;"
                  type="text"
                  value="Test Name"
                />
              </div>
            </div>
          </div>
          <div
            class="form__row"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-3 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                Номер карты
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="cardNumber"
                  type="text"
                  value="1231 2312 3123 2131"
                />
              </div>
            </div>
          </div>
          <div
            class="form__row"
          >
            <div
              class="form__col"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-3"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  MM/YY
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                    name="expiryDate"
                    type="text"
                    value="11/22"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="form__col"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-3 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  CVC
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    name="cvc"
                    type="text"
                    value="123"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="form__cardWrapper"
        >
          <div
            class="card"
          >
            <div
              class="card__top"
            >
              <img
                alt="лого"
                class="card__logo"
                src="card-logo.svg"
              />
              <p
                class="card__date"
              >
                11/22
              </p>
            </div>
            <p
              class="card__number"
            >
              <span>
                1231
              </span>
              <span>
                2312
              </span>
              <span>
                3123
              </span>
              <span>
                2131
              </span>
            </p>
            <div
              class="card__bottom"
            >
              <img
                alt="чип"
                class="card__schema"
                src="card-schema.svg"
              />
              <img
                alt="mastercard"
                class="card__mastercard"
                src="master-card.svg"
              />
            </div>
          </div>
        </div>
      </form>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="button"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Сохранить
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProfileForm component ProfileForm with card snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form__wrapper"
      >
        <h1
          class="form__title"
        >
          Профиль
        </h1>
        <p
          class="form__text"
        >
          Введите платежные данные
        </p>
        <form
          class="form__formWrapper"
          novalidate=""
        >
          <div
            class="form__form"
          >
            <div
              class="form__row"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-4 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Имя владельца
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    name="cardName"
                    style="text-transform: uppercase;"
                    type="text"
                    value="Test Name"
                  />
                </div>
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-4 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Номер карты
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    name="cardNumber"
                    type="text"
                    value="1231 2312 3123 2131"
                  />
                </div>
              </div>
            </div>
            <div
              class="form__row"
            >
              <div
                class="form__col"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-input-4"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    MM/YY
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                      name="expiryDate"
                      type="text"
                      value="11/22"
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                            />
                            <path
                              d="M0 0h24v24H0z"
                              fill="none"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="form__col"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-input-4 MuiFormControl-fullWidth"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    CVC
                  </label>
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input"
                      name="cvc"
                      type="text"
                      value="123"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form__cardWrapper"
          >
            <div
              class="card"
            >
              <div
                class="card__top"
              >
                <img
                  alt="лого"
                  class="card__logo"
                  src="card-logo.svg"
                />
                <p
                  class="card__date"
                >
                  11/22
                </p>
              </div>
              <p
                class="card__number"
              >
                <span>
                  1231
                </span>
                <span>
                  2312
                </span>
                <span>
                  3123
                </span>
                <span>
                  2131
                </span>
              </p>
              <div
                class="card__bottom"
              >
                <img
                  alt="чип"
                  class="card__schema"
                  src="card-schema.svg"
                />
                <img
                  alt="mastercard"
                  class="card__mastercard"
                  src="master-card.svg"
                />
              </div>
            </div>
          </div>
        </form>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          data-testid="button"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Сохранить
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form__wrapper"
    >
      <h1
        class="form__title"
      >
        Профиль
      </h1>
      <p
        class="form__text"
      >
        Введите платежные данные
      </p>
      <form
        class="form__formWrapper"
        novalidate=""
      >
        <div
          class="form__form"
        >
          <div
            class="form__row"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-4 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                Имя владельца
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="cardName"
                  style="text-transform: uppercase;"
                  type="text"
                  value="Test Name"
                />
              </div>
            </div>
          </div>
          <div
            class="form__row"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-4 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                data-shrink="true"
              >
                Номер карты
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  name="cardNumber"
                  type="text"
                  value="1231 2312 3123 2131"
                />
              </div>
            </div>
          </div>
          <div
            class="form__row"
          >
            <div
              class="form__col"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-4"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  MM/YY
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                    name="expiryDate"
                    type="text"
                    value="11/22"
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                          />
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="form__col"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-input-4 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
                  data-shrink="true"
                >
                  CVC
                </label>
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiInput-input"
                    name="cvc"
                    type="text"
                    value="123"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="form__cardWrapper"
        >
          <div
            class="card"
          >
            <div
              class="card__top"
            >
              <img
                alt="лого"
                class="card__logo"
                src="card-logo.svg"
              />
              <p
                class="card__date"
              >
                11/22
              </p>
            </div>
            <p
              class="card__number"
            >
              <span>
                1231
              </span>
              <span>
                2312
              </span>
              <span>
                3123
              </span>
              <span>
                2131
              </span>
            </p>
            <div
              class="card__bottom"
            >
              <img
                alt="чип"
                class="card__schema"
                src="card-schema.svg"
              />
              <img
                alt="mastercard"
                class="card__mastercard"
                src="master-card.svg"
              />
            </div>
          </div>
        </div>
      </form>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        data-testid="button"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Сохранить
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
